pkg/resources/user_acceptance_test.go

```diff
@@ -572,3 +577,219 @@ func TestAcc_User_AllParameters(t *testing.T) {
 		},
 	})
 }
+
+func TestAcc_User_issue2836(t *testing.T) {
+	userId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	defaultRole := "SOME ROLE WITH SPACE case sensitive"
```

Comment:
Isn't it better to create random identifier with this as prefix? just to be sure this test won't fail once and then start failing with "this role already exists".

Reply:
No role is created in this test because default_x is not validated with CREATE/ALTER USER (check the docs here: https://docs.snowflake.com/en/sql-reference/sql/create-user#optional-object-properties-objectproperties, e.g. `In addition, the CREATE USER operation does not verify that the role exists.`).